File: javascript/json-transform/src/functions/TransformerFunctionOr.ts
import TransformerFunction from "./common/TransformerFunction";
import FunctionContext from "./common/FunctionContext";
import { isTruthy } from "../JsonHelpers";

class TransformerFunctionOr extends TransformerFunction {
  constructor() {
    super({});
  }

  override async apply(context: FunctionContext): Promise<any> {
    const value = await context.getJsonElementStreamer(null);
    return value?.stream().any(isTruthy);
  }
}

export default TransformerFunctionOr;

File: java/json-transform/src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionOr.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.functions.common.FunctionContext;
import co.nlighten.jsontransform.functions.common.TransformerFunction;

public class TransformerFunctionOr extends TransformerFunction {
    public TransformerFunctionOr() {
        super();
    }
    @Override
    public CompletionStage<Object> apply(FunctionContext context) {
        var value = context.getJsonElementStreamer(null);
        var adapter = context.getAdapter();
        return value.stream().anyMatch(adapter::isTruthy);
    }
}
